File: app/src/main/java/ardjomand/leonardo/nutrimeal/auth/account/AccountContract.java
package ardjomand.leonardo.nutrimeal.auth.account;

public interface AccountContract {

    interface Presenter {

        void setView(View view);

        void createAccount(String name, String email, String password, String repeatedPassword, String type);

        void getCurrentUser();
    }

    interface View {

        void showProgressBar(boolean visibility);

        void goToMainScreen();

        void showError(String message);
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/auth/account/AccountContract.kt
package ardjomand.leonardo.nutrimeal.auth.account

internal interface AccountContract {

    interface Presenter {

        fun setView(view: View?)

        fun createAccount(name: String,
                          email: String,
                          password: String,
                          repeatedPassword: String,
                          type: String)

        fun getCurrentUser()
    }

    interface View {

        fun showProgressBar(visibility: Boolean)

        fun goToMainScreen()

        fun showError(message: String)
    }
}
